packages/react-composites/src/composites/common/Drawer/MoreDrawer.tsx

```diff
@@ -230,6 +261,137 @@ export const MoreDrawer = (props: MoreDrawerProps): JSX.Element => {
     });
   }
 
+  //Captions drawer menu
+
+  /* @conditional-compile-remove(close-captions) */
+  const startCaptionsButtonProps = useAdaptedSelector(_startCaptionsButtonSelector);
+  /* @conditional-compile-remove(close-captions) */
+  const startCaptionsButtonHandlers = useHandlers(_StartCaptionsButton);
+  /* @conditional-compile-remove(close-captions) */
+  const changeSpokenLanguageProps = useAdaptedSelector(_changeSpokenLanguageSelector);
+  /* @conditional-compile-remove(close-captions) */
+  const changeSpokenLanguageHandlers = useHandlers(_CaptionsSettingModal);
+  /* @conditional-compile-remove(close-captions) */
+  const [isSpokenLanguageDrawerOpen, setIsSpokenLanguageDrawerOpen] = useState<boolean>(false);
+  /* @conditional-compile-remove(close-captions) */
+  const [currentSpokenLanguage, setCurrentSpokenLanguage] = useState<string>(
+    !startCaptionsButtonProps.currentSpokenLanguage || startCaptionsButtonProps.currentSpokenLanguage === ''
+      ? 'en-us'
+      : startCaptionsButtonProps.currentSpokenLanguage
+  );
+  /* @conditional-compile-remove(close-captions) */
+  const spokenLanguageDrawerItems: DrawerMenuItemProps[] | undefined = props.isCaptionsSupported
+    ? changeSpokenLanguageProps?.supportedSpokenLanguages?.map((language) => ({
+        itemKey: language,
+        text: language,
+        onItemClick: () => {
+          setCurrentSpokenLanguage(language);
+        },
+        secondaryIconProps: currentSpokenLanguage === language ? { iconName: 'Accept' } : undefined
+      }))
+    : undefined;
+  /* @conditional-compile-remove(close-captions) */
+  const spokenLanguageDrawerStyles: _DrawerMenuStyles | undefined = props.isCaptionsSupported
+    ? {
+        root: {
+          height: _pxToRem(300),
+          overflow: 'scroll'
+        },
+        drawerSurfaceStyles: {
+          drawerContentContainer: {
+            root: {
+              span: {
+                fontWeight: 600,
+                fontSize: _pxToRem(17),
+                lineHeight: _pxToRem(22),
+                color: theme.palette.neutralDark
+              }
+            }
+          }
+        }
+      }
+    : undefined;
+  /* @conditional-compile-remove(close-captions) */
+  if (props.isCaptionsSupported) {
+    const captionsDrawerItems: DrawerMenuItemProps[] = [];
+
+    const themedToggleButtonStyle = (theme: Theme, checked: boolean): Partial<IToggleStyles> => {
+      if (checked) {
+        return {
+          root: {
+            margin: 0
+          },
+          pill: {
+            backgroundColor: `${theme.palette.themePrimary} !important`
+          },
+          thumb: {
+            backgroundColor: `${theme.palette.white} !important`
+          }
+        };
```

Comment:
Let's move all this style into a style file? Maybe MoreDrawer.style.ts or something